### PR TITLE
Update types for Leaflet v1.4 and v1.3 

### PR DESCRIPTION
**Pull request by ZhangYiJiang**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Leaflet/Leaflet/releases
    - Add autoPan to marker options - added in v1.3.0. 
    - Add closeOnEscapeKey option on Popup - added in v1.3.0- 
    - Allow crossOrigin to be a string - added in v1.3.2
    - Allow dashArray to be an array of numbers - added in v1.3.2
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/leaflet/index.d.ts [hunk 1]
```diff
@@ -626,7 +626,7 @@ export interface PathOptions extends InteractiveLayerOptions {
     opacity?: number;
     lineCap?: LineCapShape;
     lineJoin?: LineJoinShape;
-    dashArray?: string;
+    dashArray?: string | number[];
```
Comment (amcasey) [inline types/leaflet/index.d.ts]: I had a quick look at the underlying library and it didn't obviously support non-string values for `dashArray`.  Do you have a link to the code that's consuming an array instead?

File: types/leaflet/index.d.ts [hunk 2]
```diff
@@ -1534,6 +1535,9 @@ export interface MarkerOptions extends InteractiveLayerOptions {
     opacity?: number;
     riseOnHover?: boolean;
     riseOffset?: number;
+    autoPan?: boolean;
```
Comment (amcasey) [inline types/leaflet/index.d.ts]: It might be nice to add tests covering the new properties.

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/leaflet/index.d.ts (hunk 1 above)
Comment (ZhangYiJiang) [reply types/leaflet/index.d.ts]: I'm looking at this PR https://github.com/Leaflet/Leaflet/pull/6200, which is part of Leaflet 1.3.2. The documentation does not appear to be updated.

File: types/leaflet/index.d.ts (hunk 2 above)
Comment (ZhangYiJiang) [reply types/leaflet/index.d.ts]: Done

File: types/leaflet/index.d.ts (hunk 1 above)
Comment (amcasey) [reply types/leaflet/index.d.ts]: Perfect.  Thanks!

**Review by amcasey — APPROVED** [review]

**Pull request merged**